jaxdecomp/_src/fft.py

```diff
@@ -257,11 +267,20 @@ def infer_sharding_from_operands(fft_type, adjoint, mesh, arg_shapes,
     """
   # only one operand is used in pfft
   input_sharding = arg_shapes[0].sharding
-  return to_named_sharding(input_sharding)
+  if get_axis_size(input_sharding, 0) == 1 and fft_type == FftType.FFT:
+    assert (yz_slab),\
+            "If the first dimension is 1, the yz_slab flag must be set"
```

Comment:
Remove parenthesis

```suggestion
    assert yz_slab, "yz_slab flag should be set if the first dimension is 1"
```

Reply:
I also add a quick explanation for the assert.
In `infer_sharding_from_operands` we get the original sharding and pdims .
At this point and only this point I can know if the user is requesting a YZ Slab or XY Slab
In partition the pdims are (4 , 1) for example in case XY slab (because double transpose) but also (4 , 1) for YZ Slabs after the single transpose.
I impose this flag to clear this issue